package/MDAnalysis/analysis/distances.py

```diff
@@ -51,15 +51,9 @@
 import logging
 logger = logging.getLogger("MDAnalysis.analysis.distances")
 
-try:
-   from scipy import sparse
-except ImportError:
-   sparse = None
-   msg = "scipy.sparse could not be imported: some functionality will " \
-         "not be available in contact_matrix()"
-   warnings.warn(msg, category=ImportWarning)
-   logger.warn(msg)
-   del msg
+# Optional and/or lazily imported modules
+from MDAnalysis.lib import lazy
+scipy = lazy.import_module('scipy.sparse', level='base')
```

Comment:
This will be equivalent to
`import scipy.sparse`
in that it results on the 'scipy' name being added to the namespace.
On module access both `scipy` and `scipy.sparse` are fully imported.

The `level='base'` keyword tells the function to return a reference to the base module (`scipy`). From then on you use `scipy.sparse.whatever`, just like it had been imported with `import`.

The difference to the default `import_module` call (without `level='base'`) is that in the default case the returned reference is to the submodule. In that case you use `sparse = import_module('scipy.sparse')`, as equivalent to `from scipy import sparse`.